packages/date-picker/src/vaadin-date-picker-overlay-content-mixin.js

```diff
@@ -931,7 +949,8 @@ export const DatePickerOverlayContentMixin = (superClass) =>
 
     /** @private */
     _focusAllowedDate(dateToFocus, diff, keepMonth) {
-      if (this._dateAllowed(dateToFocus)) {
+      // For this check we do consider the isDateDisabled function because disabled dates are allowed to be focused, just not outside min/max
+      if (this._dateAllowed(dateToFocus, undefined, undefined, () => false)) {
```

Comment:
In this one place we want to bypass any custom `isDateDisabled` function since we are allowed to focus the date, just not select it.